core/src/main/java/org/apache/iceberg/SnapshotRefParser.java

```diff
@@ -85,4 +84,13 @@ public static SnapshotRef fromJson(JsonNode node) {
         .maxRefAgeMs(maxRefAgeMs)
         .build();
   }
+
+  static SnapshotRefType fromType(String snapshotRefType) {
```

Comment:
Is this the right place for this? Why not `SnapshotRefType.fromString` or something similar? I think it would be more obvious to put it there since it doesn't really need to be in the ref parser.

Reply:
sgtm. I'll make sure we do the same for other enums that are affected